processing/src/main/java/org/apache/druid/query/rowsandcols/RowsAndColumns.java

```diff
@@ -112,4 +131,66 @@ static AppendableRowsAndColumns expectAppendable(RowsAndColumns input)
    */
   @Nullable
   <T> T as(Class<T> clazz);
+
+  /**
+   * Serializer for {@link RowsAndColumns} by converting the instance to {@link FrameRowsAndColumns}
+   */
+  class RowsAndColumnsSerializer extends StdSerializer<RowsAndColumns>
+  {
+    public RowsAndColumnsSerializer()
+    {
+      super(RowsAndColumns.class);
+    }
+
+    @Override
+    public void serialize(
+        RowsAndColumns rac,
+        JsonGenerator jsonGenerator,
+        SerializerProvider serializerProvider
+    ) throws IOException
+    {
+      FrameRowsAndColumns frameRAC = rac.as(FrameRowsAndColumns.class);
+      if (frameRAC == null) {
+        throw DruidException.defensive("Unable to serialize RAC");
+      }
+      JacksonUtils.writeObjectUsingSerializerProvider(jsonGenerator, serializerProvider, frameRAC.getSignature());
+
+      Frame frame = frameRAC.getFrame();
+      final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+      frame.writeTo(
+          Channels.newChannel(baos),
+          false,
+          ByteBuffer.allocate(Frame.compressionBufferSize((int) frame.numBytes())),
```

Comment:
`compression` is false; meanwhile a `compressionBuffer` is being allocated at every call - is that required?
if its needed - would it be possible to reuse the buffer later?

Reply:
Since we are not trying to compress, I made the buffer `null`